tools/installer/exaile.spec

```diff
@@ -27,87 +27,85 @@ binaries = []
 # packaging's subpackages.
 hiddenimports += collect_submodules('packaging')
 
-datas =[
-  ('_inst/usr/share/exaile/data', 'data'),
-  ('_inst/usr/share/exaile/plugins', 'plugins'),
-  ('_inst/usr/share/locale', 'share/locale')
+datas = [
+    ('_inst/usr/share/exaile/data', 'data'),
+    ('_inst/usr/share/exaile/plugins', 'plugins'),
+    ('_inst/usr/share/locale', 'share/locale'),
+]
+
+# requires https://github.com/pyinstaller/pyinstaller/pull/7040 (PyInstaller >= 5.4)
+gst_exclude_plugins = [
+    'aom',
+    'assrender',
+    'cacasink',
+    'daala',
+    'dvdread',
+    'dvdsub',
+    'faac',
+    'mxf',
+    'openal',
+    'openexr',
+    'opengl',
+    'openh264',
+    'opencv',
+    'resindvd',
+    'rtmp',
+    'schro',
+    'video',
+    'vpx',
+    'wasapi',  # Generally buggy, e.g. https://github.com/exaile/exaile/issues/532
+    'webp',
+    'webrtc',
+    'x264',
+    'x265',
+    'xvimage',
+    'zbar',
 ]
 
-# requires https://github.com/pyinstaller/pyinstaller/pull/3608
-def assemble_hook(analysis):
-    # filter out gstreamer plugins we don't want
-    to_remove = [
-      'gstaom',
-      'gstassrender',
-      'gstcacasink',
-      'gstdaala',
-      'gstdvdread',
-      'gstdvdsub',
-      'gstfaac',
-      'gstmxf',
-      'gstopenal',
-      'gstopenexr',
-      'gstopengl',
-      'gstopenh264',
-      'gstopencv',
-      'gstresindvd',
-      'gstrtmp',
-      'gstschro',
-      'gstvideo',
-      'gstvpx',
-      'gstwasapi',  # Generally buggy, e.g. https://github.com/exaile/exaile/issues/532
-      'gstwebp',
-      'gstwebrtc',
-      'gstx264',
-      'gstx265',
-      'gstxvimage',
-      'gstzbar',
-    ]
-    
-    def _exclude(b):
-        for r in to_remove:
-            if r in b and 'libgstvideo-1' not in b:
-                print("Excluding", b)
-                return True
-        return False
```

Comment:
I don't remember what this is for; is it not needed anymore?

Reply:
This is how we were removing the unwanted gstreamer plugins up until now; it is part of the `assemble_hook` function, which was executed by our modified PyInstaller before performing binary analysis.

The same can now be achieved by explicitly filtering gstreamer plugins via PyInstaller's hook configuration.